src/mame/drivers/tumbleb.cpp

```diff
@@ -690,7 +685,25 @@ void tumbleb_state::fncywld_main_map(address_map &map)
 	map(0x340400, 0x34047f).nopw(); /* Unused col scroll */
 	map(0x342000, 0x3421ff).nopw();
 	map(0x342400, 0x34247f).nopw();
+}
+
+void tumbleb_state::fncywld_main_map(address_map &map)
+{
+	map(0x000000, 0x0fffff).rom();
 	map(0xff0000, 0xffffff).ram();
+
+	unico_base_map(map);
+}
+
+
+void tumbleb_state::magipur_main_map(address_map &map)
+{
+	map(0x000000, 0x00ffff).ram().share("mainram");
+	map(0xf00000, 0xffffff).rom().region("maincpu", 0);
+
+	map(0x100010, 0x100011).nopw(); // TODO: could be some kind of banking, maybe game boots with ROM at 0 instead of copying vectors, then switches it with this
```

Comment:
I would say "almost certainly" instead of "maybe." Automatically copying ROM to RAM on reset is an emulation kludge that wouldn't make sense from a hardware perspective (though quite a number of computers have bootstrap code programmed to copy itself into RAM and continue executing from there).

The exact method used to remove ROM from low memory on CPU architectures that need it there after a reset varies widely. Sometimes there's a memory address or GPIO bit dedicated to it; sometimes it's just any access to the boot ROM at its proper location; sometimes it's after the first write to a certain part of memory (e.g. sdk51). There are even some 8080 and Z80 boards which don't even map their boot ROMs at 0 for a moment, instead gating a JMP instruction onto the data bus during the first three read cycles.

Unfortunately, MAME's M68000 core is currently set up to read the initial SP and PC values at `device_reset` time, which makes proper emulation of boot ROM overlays even more awkward than it is with other CPU types.

Reply:
studying the code I'm not convinced it's this register tho, as it does seem to write this every frame (unless it has to keep writing it to maintain the mapping)

it does look like its should boot with 0x000 - 0x7ff as ROM tho, then probably switch it to RAM, because the initial RAM test bypasses that area, before it then writes the interrupt vectors.

Reply:
studying the code there really doesn't seem to be any write connected to changing between RAM and ROM, the writes to the vector table happen early on, almost immediately after the RAM test, with no register writes between so this seems unrelated